### ISSUE TITLE
Adding a standard way to get uncertainties for fitted parameters ?

### ISSUE THREAD

**Issue by saimn** [astropy/astropy#7202]
Currently getting the uncertainties for the fitted parameters is not obvious. One has to use `fitter.fit_info['param_cov']` but the order of parameters if not obvious: it is the same as `model.param_names` but fixed parameters must be removed. So I currently use something like this to create a model object containing the errors (using `_fitter_to_model_params` to filter fixed params):
```python
err_params = np.sqrt(np.diag(fitter.fit_info['param_cov']))
err = model.copy()
fitting._fitter_to_model_params(err, err_params)
```

There is already #4521 and #5197 about the documentation aspect, but It would be good to have a more user-friendly and standard way to get the errors. One idea could be to get the error as an attribute of the `Parameter` objects ? (`model.<param>.error` or `model.<param>.uncertainty` ?)

cc @nden and @eteq @mirca @larrybradley as this could be of interest for photutils.

**Comment by larrybradley** [comment]
I like this idea, but we should also keep (or have an easy way to get) the full covariance matrix (e.g. to see if the parameters are correlated).

**Comment by astrofrog** [comment]
What about having a method that takes a list of parameters and returns a covariance matrix? Then one can easily get the specific subset of the full covariance matrix one is interested in.

**Comment by karllark** [comment]
For uncertainties, the following are desired.  Depending if the fitter can provide the information.

uncertainty on each fit parameter (1 sigma, assuming Gaussian)
plus/minus uncertainty on each fit parameter (+ 1 sigma, -1 sigma assuming Gaussian)
covariance matrix (assumes a multivariate Gaussian)
samples of nD probability (MCMC samples, bootstrap/jackknife samples)
1D probabilities (marginalized pPDF from MCMC samples)

**Comment by tepickering** [comment]
the method to get parameter uncertainties in the original comment currently works only with ```LevMarLSQFitter```. parameter covariance matrices are not returned by the other fitters.

**Comment by karllark** [comment]
A proposed implementation would be:

where uncertainty is an object that contains the above info as
g = model
param = specific parameter

g.param.unc
g.param.plus_unc
g.param.minus_unc
g.param.1d_probability

g.covariance_matrix
g.probability_samples

Any of the above uncertainties that are not supplied or cannot be computed from the fitter results should be set to None.

The fitter/optimizer that was used should be returned as part of the fitted model (an attribute).

**Comment by tepickering** [comment]
i do think uncertainties here should use ```astropy.uncertainty``` as much as possible to avoid wheel reinvention. see also discussion in #9020 and #9044.

**Comment by karllark** [comment]
I'm not sure that the uncertainty on a parameter from a fitter should be *only* a distribution.  This makes is harder for the usual and likely very dominate case where the user just wants to know the 1-sigma uncertainty assuming a Gaussian model on a parameter.  Maybe having a separate bit of info that is unc_function would allow for the distribution info as well as the "standard" uncertainty.

**Comment by tepickering** [comment]
yes, the ```Distribution``` classes that are developed should have handy convenience methods for things like ```.value```, ```.sigma```, ```.covariance```, and so on.
